Improve modules promethues readme (#10434)
emqx/emqx: apps/emqx_modules/README.md

# EMQX Modules

> The application provide some minor functional modules that are not included in the MQTT

Reviewer: ```suggestion
The application provides some minor functional modules that are not included in the MQTT
```

protocol standard, including "Delayed Publish", "Topic Rewrite", "Topic Metrics" and "Telemetry".

## Delayed Publish

After enabling this module, messages sent by the clients with the topic prefixed with
`$delayed/{Interval}/{Topic}` will be delayed by `{Interval}` seconds before
being published to the `{Topic}`.

More introduction about [Delayed Publish](https://www.emqx.io/docs/en/v5.0/mqtt/mqtt-delayed-publish.html).

See [Enabling/Disabling Delayed Publish via HTTP API](https://www.emqx.io/docs/en/v5.0/admin/api-docs.html#tag/MQTT/paths/~1mqtt~1delayed/put).

## Topic Rewrite

Topic Rewrite allows users to configure rules to change the topic strings that
the client requests to subscribe or publish.

> This feature is very useful when it need to compatibility with different versions of topic designs.

Reviewer 2: Perhaps:

```suggestion
This feature is very useful when there's a need to transform between different topic structures.
```
?

# EMQX Modules

The application provide some minor functional modules that are not included in the MQTT
protocol standard, including "Delayed Publish", "Topic Rewrite", "Topic Metrics" and "Telemetry".

## Delayed Publish

> After enabling this module, messages sent by the user with the prefix

Reviewer 3: This module provides a feature that allows for manual intervention in the publishing delay of messages in an MQTT system.
After the module is enabled, messages sent by the MQTT clients with the topic

> nit: only mqtt clinets? how to delay a message for gateway clinets?

> After enabling this module, messages sent by the user with the prefix

Author: ```suggestion
After enabling this module, messages sent by the clients with the topic prefixed with
```

`$delayed/{Interval}/{Topic}` will be delayed by `{Interval}` seconds before
being published to the `{Topic}`.

More introduction about [Delayed Publish](https://www.emqx.io/docs/en/v5.0/mqtt/mqtt-delayed-publish.html).

See [Enabling/Disabling Delayed Publish via HTTP API](https://www.emqx.io/docs/en/v5.0/admin/api-docs.html#tag/MQTT/paths/~1mqtt~1delayed/put).

## Topic Rewrite

Topic Rewrite allows users to configure rules to change the topic strings that
the client requests to subscribe or publish.

> This feature is very useful when designing topics that are compatible with different
> client versions. For example, an old device that has already been issued and cannot

Author: ```suggestion
This feature is very useful when it need to compatibility with different versions of topic designs.
For example, an old device that has already been issued and cannot
```

be upgraded may use old topic rules, but the production environment need to apply

> a new design rules for the topics.

Reviewer 3: This feature is useful to designing compatible topic system with different versions of client implements.
For example, a device will pub/sub message with topic `old/topic` with previous client system implement.
But the new devices implement doesn't contain the topic.
At this point, the topic-rewrite feature can be used to map the topics used by old devices to the topics used by the new system, achieving compatibility between new and old devices within the same system.

> be upgraded may use old topic rules, but the production environment need to apply
> a new design rules for the topics.

Author: ```suggestion
be upgraded may use old topic rules, but for some reason, we adjusted the format of topics. We can use this feature to rewrite the old topics as the new formats to eliminate this difference.
```
